### PR TITLE
Deprecate timeout, refactor Submitter.publish_job

### PR DESCRIPTION
**Pull request by darkside**
Changes:
- Deprecated timeout feature until we can figure out how to do timeout in a thread safe way
- Changed Submitter.publish_job to handle its connection if it isn't connected yet
- Version bump

Note: I had to lockdown bunny to before 2.0 as it was giving us weird channel errors in IH production.

### CODE REVIEW

File: coney_island.gemspec [hunk 1]
```diff
@@ -21,7 +21,7 @@ Gem::Specification.new do |s|
 
   s.add_dependency "rails", ">= 4.0.1"
   s.add_dependency "amqp", ">= 1.5.0"
-  s.add_dependency "bunny", ">= 1.7.0"
+  s.add_dependency "bunny", "< 2.0"
```
Comment (edraut) [inline coney_island.gemspec]: didn't this cause other errors in our loader.io tests?

**Comment by edraut** [comment]
@darkside Looks good other than that bunny version bump.

File: coney_island.gemspec (hunk 1 above)
Comment (darkside) [inline coney_island.gemspec]: We are locking it down so it *doesn't* upgrade, the 2.0 version was causing random channel errors.

Comment (edraut) [inline coney_island.gemspec]: Oh! I see now. I missed the `<` Awesome.

**Comment by edraut** [comment]
:+1: 
Let 'er rip

**Pull request merged**